src/tools/chip-cert/tests/test_chip_cert.py

```
@@ -0,0 +1,282 @@
+
+    @classmethod
+    def setUpClass(cls):
+        from_env = os.environ.get("CHIP_CERT")
+        if not from_env:
+            raise AssertionError("CHIP_CERT is not set; point it at the chip-cert binary to test")
+        cls.chip_cert = Path(from_env)
+        if not cls.chip_cert.is_file():
+            raise AssertionError(f"chip-cert binary not found at {cls.chip_cert}")
+
+    def setUp(self):
+        directory = tempfile.TemporaryDirectory()
```

Pylint (4.0.6)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)